src/js/components/DataFilters/DataFilters.js

```diff
@@ -77,6 +76,30 @@ export const DataFilters = ({
   }, [filtersCleared]);
   const controlled = useMemo(() => drop || layer, [drop, layer]);
 
+  const configured = Children.count(children) === 0;
+  useEffect(() => {
+    // when view changes via DataView or user interacting with filters,
+    // adjust badge to reflect that
+    if (controlled && view.properties) {
+      const nextTouched = { ...view.properties };
+      Object.keys(nextTouched).forEach((k) => {
+        if (
+          properties?.[k]?.badge === false ||
+          (configured && properties && !properties?.[k])
+        ) {
+          delete nextTouched[k];
```

Comment:
Do we need to do `delete view.properties[k];` here too?

Reply:
No we don't need it here. For "non RangeSelector" cases, view already removes properties that don't have a value. So we just need this special case for RangeSelector. Although, part of me is wondering if when `onView` is called it should have already sifted the RangeSelector out if it's at its min/max... so it behaves more in alignment with the other properties. Let me explore this briefly.

Reply:
Just pushed the cleanups related to this in my latest commit.